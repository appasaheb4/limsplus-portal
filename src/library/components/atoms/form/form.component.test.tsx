import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {
  CheckBox,
  Input,
  Toggle,
  SelectOption,
  MultilineInput,
  Clock,
  Label,
  InputDate,
  InputDateTime,
  InputFile,
  InputRadio,
  InputWrapper,
} from './form.component';

describe('Checkbox component', () => {
  it('render checkbox correctly', () => {
    const checkbox = render(<CheckBox />);
    expect(checkbox).toMatchSnapshot();
  });
});

describe('Input component', () => {
  it('render input correctly', () => {
    const inputComp = render(
      <Input
        type='text'
        value='test'
        onChange={val => jest.fn()}
        defaultValue='test'
      />,
    );
    const input = inputComp.getByTestId('INPT');
    fireEvent.change(input, {target: {value: 'test'}});
    expect(input).toMatchSnapshot();
  });
});
// describe('Toggle component', () => {
//   it('render toggle correctly', () => {
//     const toggle = render(<Toggle />);
//     expect(toggle).toMatchSnapshot();
//   });
// });

// describe('SelectOption component', () => {
//   it('render selectOption correctly', () => {
//     const select = render(<SelectOption key='vd' />);
//     expect(select).toMatchSnapshot();
//   });
// });
// describe('MultilineInput component', () => {
//   it('render multilineInput correctly', () => {
//     const multi = render(<MultilineInput />);
//     expect(multi).toMatchSnapshot();
//   });
// });

// describe('Clock component', () => {
//   it('render clock correctly', () => {
//     const clock = render(<Clock />);
//     expect(clock).toMatchSnapshot();
//   });
// });
// describe('Label component', () => {
//   it('render label correctly', () => {
//     const label = render(<Label htmlFor='vsdvrf' />);
//     expect(label).toMatchSnapshot();
//   });
// });

// describe('InputDate component', () => {
//   it('render inputDate correctly', () => {
//     const inputDate = render(<InputDate />);
//     expect(inputDate).toMatchSnapshot();
//   });
// });
// describe('InputDateTime component', () => {
//   it('render inputDateTime correctly', () => {
//     const inputDateTime = render(<InputDateTime />);
//     expect(inputDateTime).toMatchSnapshot();
//   });
// });

// describe('InputFile component', () => {
//   it('render inputFile correctly', () => {
//     const inputFile = render(<InputFile />);
//     expect(inputFile).toMatchSnapshot();
//   });
// });
// describe('InputRadio component', () => {
//   it('render inputRadio correctly', () => {
//     const inputRadio = render(<InputRadio />);
//     expect(inputRadio).toMatchSnapshot();
//   });
// });

// describe('InputWrapper component', () => {
//   it('render inputWrapper correctly', () => {
//     const inputWrapper = render(<InputWrapper />);
//     expect(inputWrapper).toMatchSnapshot();
//   });
// });
