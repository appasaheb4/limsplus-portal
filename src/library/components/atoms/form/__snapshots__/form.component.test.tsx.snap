// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component render checkbox correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li>
        <input
          type="checkbox"
          value=""
        />
         
      </li>
    </div>
  </body>,
  "container": <div>
    <li>
      <input
        type="checkbox"
        value=""
      />
       
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input component render input correctly 1`] = `
<input
  autocomplete="given-name"
  class="undefined leading-4 p-2  focus:outline-none focus:ring  block w-full shadow-sm sm:text-base  border-2  border-gray-300 rounded-md"
  data-testid="INPT"
  type="text"
  value="test"
/>
`;
